generative_ai/context_caching/get_list_of_content_caches.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+
+from typing import List
+
+from google.cloud.aiplatform_v1beta1.types.cached_content import CachedContent
+
+PROJECT_ID = os.getenv("GOOGLE_CLOUD_PROJECT")
+
+
+def get_list_of_context_caches() -> List[CachedContent]:
+    # [START generativeaionvertexai_gemini_get_list_of_context_caches]
+    import json
+
+    from datetime import datetime as dt, timezone as tz
+
+    import vertexai
+
+    from vertexai.preview import caching
+
+    # TODO(developer): Update project id and location
+    vertexai.init(project=PROJECT_ID, location="us-central1")
+
+    cached_content_list = caching.CachedContent.list()
+    # Access individual properties of a CachedContent object
+    for cc in cached_content_list:
+        expire_time_in_seconds = int((cc.expire_time - dt.now(tz.utc)).total_seconds())
+        model_name = cc.model_name.split("/")[-1]
+        print(
+            f"Cached content '{cc.display_name}' for model '{model_name}' expires in {expire_time_in_seconds} s."
+        )
+        # Example output:
+        # Cached content 'scientific-articles' for model 'gemini-1.5-pro-001' expires in 3000 s.
+
+    # or convert the ContentCache object to a dictionary:
+    for cc in cached_content_list:
+        print(json.dumps(cc.to_dict(), indent=2))
```

Comment:
I don't know if there's a reason to change this to a dictionary. Devs can use the `CachedContent` class

Reply:
@holtskinner , no doubts they can! But I also wanted to show them an alternative, that the object also serializable as a dictionary. Paired with the [example output](https://github.com/GoogleCloudPlatform/python-docs-samples/pull/12596/files#diff-fe795838a2d14ddf893e452ae6d27a3873cb08355be46cc55808aa730419d376R50) comment below, it provides more context about what you can get from the object. I'm simply trying to look at the docs as a total Vertex newcomer (which I am!) and incorporate some bits of context that significantly overlap with the main focus of a sample. I'm happy to re-consider and remove this bit if it's not helpful.